packages/mui-base/src/unstable_useModal/useModal.ts

```diff
@@ -135,7 +135,11 @@ export function useModal(parameters: UseModalParameters): UseModalReturnValue {
     // clicking a checkbox to check it, hitting a button to submit a form,
     // and hitting left arrow to move the cursor in a text input etc.
     // Only special HTML elements have these default behaviors.
-    if (event.key !== 'Escape' || !isTopModal()) {
+    if (
+      event.key !== 'Escape' ||
+      event.which === 229 || // Wait until IME is settled.
```

Comment:
`.which` is deprecated. Could we use `.isComposing` instead?

Reply:
FYI I found in an old issue about `isComposing` with an odd behavior in macOS Safari: https://github.com/mui/material-ui/issues/19435#issuecomment-579586245

Reply:
As per the issue comment above, it needs to use `which` instead of `isComposing` for Safari.
[This behavior is not fixed on the latest (17.1) Safari version.](https://bugs.webkit.org/show_bug.cgi?id=165004)